### ISSUE TITLE
AutoDock GPU Docking Result Unreproducible

### ISSUE THREAD

**Issue by jkslionsss** [ccsb-scripps/AutoDock-GPU#281]
I performed docking using the same map.fld file and ligand.pdbqt file with the following command:

/opt/vina_gpu/AutoDock-GPU-1.5.3/bin/autodock_gpu_128wi --seed 12345,12345,12345 --ffile 1qf1_receptor.maps.fld --lfile 1qf1_ligand.pdbqt --npdb 1 --nrun 1 --devnum 1

However, I noticed that the results are different. Could this indicate a lack of reproducibility?

![image](https://github.com/user-attachments/assets/c1271ef7-3a3b-49b3-8782-eff14e0cafe9)

![image](https://github.com/user-attachments/assets/90db8a68-64c1-44d1-b25c-3f09fd516ebd)

![image](https://github.com/user-attachments/assets/45cc6732-f6b5-491d-ac8e-abe99dcc3a29)

Additionally, I am attaching the map.fld file as well

![image](https://github.com/user-attachments/assets/39e16764-3120-40ce-83e0-8d1c5a354699)

**Comment by atillack** [comment]
@jkslionsss Yes and no - the random number generator used is fully reproducible and so are the starting configurations. Once on GPU though how the sequence of random numbers is used is generally not reproducible anymore when `NUMWI` is greater than 1. Technically, when NUMWI is below or equal to the actual number of concurrent work items on device runs will be reproducible on the same device - but results will still be different for different NUMWI values (just the same for the same NUMWI when below the device limit). Long story short, `--seed` is good for debugging in certain circumstances but it's not true reproducibility in all circumstances.
